Scripts/Runtime/Jobs/Collections/AccessController.cs

```diff
@@ -0,0 +1,141 @@
+        {
+            //No one else can write until this job is done
+            WriteAccessDependency = JobHandle.CombineDependencies(WriteAccessDependency, jobThatWrites);
+            //No one else can do anything until this job is done writing
+            ParallelWriteAccessDependency = JobHandle.CombineDependencies(ParallelReadAccessDependency, jobThatWrites);
+            ReadAccessDependency = JobHandle.CombineDependencies(ReadAccessDependency, jobThatWrites);
+            ParallelReadAccessDependency = JobHandle.CombineDependencies(ParallelReadAccessDependency, jobThatWrites);
+        }
+        
+        /// <summary>
+        /// Ensures that the <see cref="AccessController"/> is aware of a job that is writing in parallel.
+        /// </summary>
+        /// <param name="jobThatWritesInParallel">The job that writes in parallel.</param>
+        public void AddJobHandleForParallelWriting(JobHandle jobThatWritesInParallel)
+        {
+            ParallelWriteAccessDependency = JobHandle.CombineDependencies(ParallelWriteAccessDependency, WriteAccessDependency, jobThatWritesInParallel);
+            //No one else can read until all writing is complete
+            ReadAccessDependency = JobHandle.CombineDependencies(ReadAccessDependency, jobThatWritesInParallel);
+            ParallelReadAccessDependency = JobHandle.CombineDependencies(ParallelReadAccessDependency, jobThatWritesInParallel);
+        }
+        
+        /// <summary>
+        /// Ensures that the <see cref="AccessController"/> is aware of a job that is reading.
+        /// </summary>
+        /// <param name="jobThatReads">The job that reads.</param>
+        public void AddJobHandleForReading(JobHandle jobThatReads)
+        {
+            ReadAccessDependency = JobHandle.CombineDependencies(ReadAccessDependency, jobThatReads);
+            //No one else can write until we're done reading
+            WriteAccessDependency = JobHandle.CombineDependencies(WriteAccessDependency, jobThatReads);
+            ParallelWriteAccessDependency = JobHandle.CombineDependencies(ParallelReadAccessDependency, jobThatReads);
+        }
+
+        /// <summary>
+        /// Ensures that the <see cref="AccessController"/> is aware of a job that is reading in parallel.
+        /// </summary>
+        /// <param name="jobThatReadsInParallel">The job that reads in parallel.</param>
+        public void AddJobHandleForParallelReading(JobHandle jobThatReadsInParallel)
+        {
+            ParallelReadAccessDependency = JobHandle.CombineDependencies(ParallelReadAccessDependency, ReadAccessDependency, jobThatReadsInParallel);
+            //No one else can write until we're done reading
+            WriteAccessDependency = JobHandle.CombineDependencies(WriteAccessDependency, jobThatReadsInParallel);
+            ParallelWriteAccessDependency = JobHandle.CombineDependencies(ParallelReadAccessDependency, jobThatReadsInParallel);
+        }
```

Comment:
Left a few suggestions. One style suggestion that could be applied to every case and one opportunity to do a direct assignment instead of combining (I think)

Reply:
Made the changes and tested them to be sure and they work as expected. Thanks!